src/tensor_product_operators.jl

```diff
@@ -0,0 +1,168 @@
+# We use an own structure for tensor product operators to be able to specialize some functions.
+"""
+    TensorProductOperator{Dim,T}
+
+A tensor product operator is a [`MultidimensionalMatrixDerivativeOperator`](@ref) that is the tensor product of one-dimensional derivative operators.
+
+See also [`tensor_product_operator_2D`](@ref).
+
+References:
+- Sigrun Ortleb (2021)
+  Numerical Methods for Fluid Flow:
+    High Order SBP Schemes, IMEX Advection-Diffusion Splitting and Positivity Preservation for Production-Destruction-PDEs
+  Habilitation thesis, University of Kassel, [DOI: 10.17170/kobra-202301037274](https://doi.org/10.17170/kobra-202301037274),
+  Chapter 1.2.4.
+- Magnus Svärd, Jan Nordström (2014)
+  Review of summation-by-parts schemes for initial-boundary-value problems
+  Journal of Computational Physics 268, pp. 17-38, [DOI: 10.1016/j.jcp.2014.02.031](https://doi.org/10.1016/j.jcp.2014.02.031).
+"""
+@auto_hash_equals struct TensorProductOperator{Dim,T} <: AbstractMultidimensionalMatrixDerivativeOperator{Dim,T}
+    D_multi::MultidimensionalMatrixDerivativeOperator{Dim,T}
+    N_x::Int
+    N_y::Int
+end
+
+# This allows us to treat a `TensorProductOperator` as a `MultidimensionalMatrixDerivativeOperator`.
+function Base.getproperty(D::TensorProductOperator, name::Symbol)
+    if name in [:D_multi, :N_x, :N_y]
```

Comment:
```suggestion
    if name in (:D_multi, :N_x, :N_y)
```
This avoids an unnecessary allocation